src/ngx_http_lua_shdict.c

```diff
@@ -55,6 +55,49 @@ static ngx_inline ngx_shm_zone_t *ngx_http_lua_shdict_get_zone(lua_State *L,
 #define NGX_HTTP_LUA_SHDICT_LEFT        0x0001
 #define NGX_HTTP_LUA_SHDICT_RIGHT       0x0002
 
+ngx_inline void
+ngx_http_lua_shdict_rlock(ngx_http_lua_shdict_ctx_t *ctx)
+{
+    ngx_rwlock_rlock(&ctx->rwlock);
+}
+
+ngx_inline void
+ngx_http_lua_shdict_wlock(ngx_http_lua_shdict_ctx_t *ctx)
+{
+    ngx_rwlock_wlock(&ctx->rwlock);
+}
+
+ngx_inline void
+ngx_http_lua_shdict_rwunlock(ngx_http_lua_shdict_ctx_t *ctx)
+{
+    ngx_rwlock_unlock(&ctx->rwlock);
+}
+
+ngx_inline void
+ngx_http_lua_shdict_rwdowngrade(ngx_http_lua_shdict_ctx_t *ctx)
+{
+    ngx_rwlock_downgrade(&ctx->rwlock);
+}
+
+ngx_inline void
+ngx_http_lua_shdict_shmtxlock(ngx_http_lua_shdict_ctx_t *ctx)
+{
+    ngx_shmtx_lock(&ctx->shpool->mutex);
+}
+
+ngx_inline void
+ngx_http_lua_shdict_shmtxunlock(ngx_http_lua_shdict_ctx_t *ctx)
+{
+    ngx_shmtx_unlock(&ctx->shpool->mutex);
+}
+
+ngx_inline void
```

Comment:
The `ngx_inline` flag is useless. These functions are only called as function pointers where inlining is impossible anyway. Am I missing anything here?